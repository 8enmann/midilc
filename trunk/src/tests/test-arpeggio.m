/* Test arpeggio
   Tests creating an arpeggio (sequence) from a chord.  Library function.
   @author Fred
    */

main(){
  Note a;
  Chord c;
  Sequence s;
  Number i;
  
  c = new_chord(C,E,G);
  s = arpeggiate(c);
  play(s);
}

arpeggiate(c)
{
    Number n;
	Number i;
	Sequence s;
	s = new_sequence();
	n = c.length;
	for(i = 0; i < n; i=i+1)
	{
		s = s + c[i];
	}
	return s;
}
