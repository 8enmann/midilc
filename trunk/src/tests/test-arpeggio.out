0,4,60
4,4,64
8,4,67

